app/models/vim_performance_daily.rb

```diff
@@ -33,11 +33,7 @@ def self.find_entries(ext_options)
   end
 
   def self.process_hourly_for_one_day(recs, options = {})
-    return [] if recs.blank?
```

Comment:
recs is an association.
No reason for a count(*) here.
The `each` loop will fetch the results soon enough.
If you note, this check was moved until right after the `each` loop.

BUT, there is no reason to call that first code here. and that `select {}` doesn't make sense since it is in the query, so I'll move that.